ocaml/lib/snarky_bindings.ml

```diff
@@ -178,6 +178,15 @@ module Gates = struct
                    { kind = Zero; values = [| in1; in2; out |]; coeffs = [||] }
                 )
           } )
+
+  let basic sl l sr r so o sm sc =
+    Impl.with_label "generic_gate" (fun () ->
+        Impl.assert_
+          { annotation = Some __LOC__
+          ; basic =
+              Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint.T
+                (Basic { l = (sl, l); r = (sr, r); o = (so, o); m = sm; c = sc })
+          } )
```

Comment:
I exposed `Basic` and configured the gate in TS insetead of implementing `and_equation` as in the [OCaml code](https://github.com/MinaProtocol/mina/blob/d5ae586319a9d0fdfd322b2b8b54beb5bfaeddbb/src/lib/crypto/kimchi_backend/gadgets/bitwise.ml#L476).

1. Is this the right approach?
1. Are the argument names appropriate?
1. Is there anything else I should expose while I'm here?

Reply:
1a) Yes it's the right approach to expose the generic gate in it's most general form and not specialized to the "and" logic. We definitely want this in o1js, out of principle!
1b) I don't think we need it for the "and" gadget though. o1js creates the right generic gates when you do `.add()` and `mul()` and so on, which we could use for "and" (will comment on the other PR)
2) I think so. I guess `sl` means "the selector for the left input" etc?
3) No I think this is good, although I think we should out of principle make an effort to expose _all_ gates to o1js, but that can be a separate PR

Reply:
1b) Ahhhh! My assumption was that that would be less efficient than doing what I did, but I think I understand now. I gather from Anaïs that when I call `.add()`, it goes on to a stack, and then once there is enough on the stack for us to generate an efficient constraint we do so. Is that correct? (Don't worry about tying it all out here if not; maybe we can just do a short call sometime and I will make a list of questions).
2) Correct!

Reply:
Yeah what Anais wrote is true but she probably didn't know that a simple `Field.mul()` also flushes that stack into a constraint, so we don't need that complicated way of specifying the gate